qiskit/circuit/quantumcircuit.py

```diff
@@ -1865,7 +1865,9 @@ def draw(
             cregbundle=cregbundle,
         )
 
-    def size(self, filter_function: Optional[callable] = lambda x: not x[0]._directive) -> int:
+    def size(
+        self, filter_function: Callable[[DataElement], bool] = lambda x: not x[0]._directive
```

Comment:
The type hints LGTM! I wonder whether it's good practice to define a lambda function in the signature? Usually we tried to avoid object construction and instead have `None` as default and only give the default value inside the function like
```python
def size(self, filter_function=None):
    if filter_function is None:
        filter_function = ....
```
Do you know whether this would be preferable here too?

Reply:
Actually, in general it is not a good idea to use lambdas, which are mutable objects, as the default argument. However, I don't see the problem anyway, so I left it as is. (Although it is mutable, it should not be changed...)

Reply:
Mutable values aren't good in defaults, but I don't think lambdas are mutable?  I guess you can change their hidden attributes if you try, but that's the same as any immutable Python object.  Certainly they're hashable, so you can put them in hash-maps (now I _really_ want to find a use for that...).

Reply:
You are right. lambda is hashable.
 <img width="396" alt="image" src="https://user-images.githubusercontent.com/6814928/169180545-45464dea-f0fc-4f5d-adc5-13cc36b74182.png">

But lambda is mutable:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/6814928/169181028-39533cac-26d8-4610-b374-280b7386e1c4.png">
So, I know this kind of code is not practical, but it is possible to put bugs in this way.

(I remember that in Python 2 it wasn't a hidden method (func_code) and I didn't care about it, but it has changed since python 3.)

Well, I guess the conclusion remains the same, that we should just leave it as it is...